sql/sqlserver/server.go

```diff
@@ -672,24 +674,38 @@ func (s *Server) processColumns(desc *structured.TableDescriptor,
 	return cols, nil
 }
 
-func (s *Server) processInsertRows(node parser.SelectStatement) (rows []sqlwire.Result_Row, err error) {
+func (s *Server) processSelect(node parser.SelectStatement) (rows []sqlwire.Result_Row, err error) {
 	switch nt := node.(type) {
+	// case *parser.Select:
+	// case *parser.Union:
+	// TODO(vivek): return s.query(nt.stmt, nil)
 	case parser.Values:
-		for _, row := range nt {
+		for _, tuple := range nt {
+			data, err := parser.EvalExpr(tuple, nil)
+			if err != nil {
+				return rows, err
+			}
+			dTuple, ok := data.(parser.DTuple)
+			if !ok {
+				// A one-element DTuple is currently turned into whatever its
+				// underlying element is, so we have to massage here.
+				// TODO(tschottdorf): Is that desired behaviour on behalf of
+				// EvalExpr?
+				dTuple = parser.DTuple([]parser.Datum{data})
+			}
 			var vals []sqlwire.Datum
-			for _, val := range row {
+			for _, val := range dTuple {
 				switch vt := val.(type) {
-				case parser.StrVal:
-					tmp := string(vt)
-					vals = append(vals, sqlwire.Datum{StringVal: &tmp})
-				case parser.NumVal:
-					tmp := string(vt)
-					vals = append(vals, sqlwire.Datum{StringVal: &tmp})
-				case parser.ValArg:
-					return rows, util.Errorf("TODO(pmattis): unsupported node: %T", val)
-				case parser.BytesVal:
-					tmp := string(vt)
-					vals = append(vals, sqlwire.Datum{StringVal: &tmp})
+				case parser.DBool:
+					vals = append(vals, sqlwire.Datum{BoolVal: (*bool)(&vt)})
+				case parser.DInt:
+					vals = append(vals, sqlwire.Datum{IntVal: (*int64)(&vt)})
+				case parser.DFloat:
+					vals = append(vals, sqlwire.Datum{FloatVal: (*float64)(&vt)})
+				case parser.DString:
+					vals = append(vals, sqlwire.Datum{StringVal: (*string)(&vt)})
+				case parser.DNull:
+					vals = append(vals, sqlwire.Datum{})
 				default:
 					return rows, util.Errorf("TODO(pmattis): unsupported node: %T", val)
```

Comment:
The only other possibility here is a `DTuple` which would be an error. Probably fine to remove the `TODO(pmattis)` now.